exs_numpy.py

```diff
@@ -0,0 +1,423 @@
+#Submitted by Keren Or Hadad 208025205
+
+import numpy as np
+#1
+x = np.arange(21)
+print("my vector:")
+print(x)
+print("After changing the sign of the numbers in the range from 9 to 15:")
+x[(x >= 9) & (x <= 15)] *= -1
+
+print(x)
+
+"""
+output:
+my vector:
+[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20]
+After changing the sign of the numbers in the range from 9 to 15:
+[  0   1   2   3   4   5   6   7   8  -9 -10 -11 -12 -13 -14 -15  16  17
+  18  19  20]
+"""
+
+#2
+
+v = np.linspace(10, 49, 5)
+print("Length 10 with values evenly distributed between 5 and 50:")
+print(v)
+
+"""
+output:
+Length 10 with values evenly distributed between 5 and 50:
+[10.   19.75 29.5  39.25 49.  ]
+"""
+
+#3
+
+m= np.arange(10,22).reshape((3, 4))
+print("Original matrix:")
+print(m)
+print("Number of rows and columns of the said matrix:")
+print(m.shape)
+
+"""
+output:
+Original matrix:
+[[10 11 12 13]
+ [14 15 16 17]
+ [18 19 20 21]]
+Number of rows and columns of the said matrix:
+(3, 4)
+"""
+#4
+
+x = np.ones((10, 10))
+x[1:-1, 1:-1] = 0
+print(x)
+
+"""
+output:
+[[1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
+ [1. 0. 0. 0. 0. 0. 0. 0. 0. 1.]
+ [1. 0. 0. 0. 0. 0. 0. 0. 0. 1.]
+ [1. 0. 0. 0. 0. 0. 0. 0. 0. 1.]
+ [1. 0. 0. 0. 0. 0. 0. 0. 0. 1.]
+ [1. 0. 0. 0. 0. 0. 0. 0. 0. 1.]
+ [1. 0. 0. 0. 0. 0. 0. 0. 0. 1.]
+ [1. 0. 0. 0. 0. 0. 0. 0. 0. 1.]
+ [1. 0. 0. 0. 0. 0. 0. 0. 0. 1.]
+ [1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]]
+"""
+
+#5
+m = np.array([[1,2,3], [4,5,6], [7,8,9], [10, 11, 12]])
+v = np.array([1, 1, 0])
+print("Original vector:")
+print(v)
+print("Original matrix:")
+print(m)
+result = np.empty_like(m)
+for i in range(4):
+  result[i, :] = m[i, :] + v
+print("\nAfter adding the vector v to each row of the matrix m:")
+print(result)
+
+"""
+output:
+Original vector:
+[1 1 0]
+Original matrix:
+[[ 1  2  3]
+ [ 4  5  6]
+ [ 7  8  9]
+ [10 11 12]]
+
+After adding the vector v to each row of the matrix m:
+[[ 2  3  3]
+ [ 5  6  6]
+ [ 8  9  9]
+ [11 12 12]]
+
+"""
+
+#6
+import matplotlib.pyplot as plt
+# Compute the x and y coordinates for points on a sine curve
+x = np.arange(0, 3 * np.pi, 0.2)
+y = np.sin(x)
+print("Plot the points using matplotlib:")
+plt.plot(x, y)
+plt.show()
+"""
+output
+Plot the points using matplotlib:
++ the image
+"""
+
+#7
+
+nums = np.arange(16, dtype='int').reshape(-1, 4)

```

Comment:
The values here are not random.
Use built numpy.random.rand
https://numpy.org/doc/stable/reference/random/generated/numpy.random.rand.html